src/lower/mod.rs

```diff
@@ -909,24 +910,42 @@ impl Anonymize for [ir::ParameterKind<ir::Identifier>] {
 }
 
 impl ir::StructDatum {
-    fn to_program_clauses(&self) -> Vec<ir::ProgramClause> {
+    fn to_program_clauses(&self, program: &ir::Program) -> Vec<ir::ProgramClause> {
         // Given:
         //
-        //    struct Foo<T: Eq> { ... }
+        //    struct Foo<T: Eq> {
+        //        field: Bar
+        //    }
         //
         // we generate the following clause:
         //
-        //    for<?T> WF(Foo<?T>) :- (?T: Eq).
+        //    for<?T> WF(Foo<?T>) :- WF(?T), WF(Bar), (?T: Eq), WF(?T: Eq).
```

Comment:
Hmm, I don't think we want to include `WF(Bar)` in this list.  That is, I think that `WF(Foo<T>)` is just saying if a **reference** to the type `Foo<T>` is well-formed. Not if the struct **definition** is well-formed -- for the latter, I would expect us to generate some rules like:

```
StructDefnWf(Foo) :-
    for<T> { if (T: Eq) { WF(Bar) } }.
```